src/hexer/xelim.nim

```diff
@@ -92,9 +92,14 @@ proc trExprInto(c: var Context; dest: var TokenBuf; n: var Cursor; v: SymId) =
   trExpr c, dest, n, tar
 
   let info = n.info
-  copyIntoKind dest, AsgnS, info:
-    dest.addSymUse v, info
+
+  var rest = beginRead(tar.t)
+  if isNoReturn(rest):
```

Comment:
Makes no sense to me. It's xelim's job to eradicate `ExprX` nodes. `isNoReturn` doesn't matter for this.

Reply:
```nim
proc foo =
  var x = if true:
      1
    else:
      return
```

xelim insert an assignment for `return` here

```
var tmp: int
if true:
  (asgn tmp 1)
else:
  (asgn tmp (ret))
```

Shouldn't we leave it alone if it is a `noreturn` node?

Reply:
That would do too. Since it only needs to check one expression, a `noreturn` test should suffice. I will try` getType` instead.